File: examples/java/Graph1.java
import java.util.*;

public class Graph1 {
    static ArrayList<ArrayList<Integer>> adj;
    static int V;

    static void addEdge(int u, int v) {
        adj.get(u).add(v);
    }

    static void bfs(int s) {
        boolean[] vis = new boolean[V];
        Queue<Integer> q = new LinkedList<>();
        q.add(s);
        vis[s] = true;
        while (!q.isEmpty()) {
            int u = q.poll();
            System.out.print(u + " ");
            for (int v : adj.get(u)) {
                if (!vis[v]) { vis[v] = true; q.add(v); }
            }
        }
    }

    public static void main(String[] args) {
        V = 4;
        adj = new ArrayList<>();
        for (int i = 0; i < V; i++) adj.add(new ArrayList<>());
        addEdge(0, 1);
        addEdge(0, 2);
        addEdge(1, 2);
        addEdge(2, 0);
        addEdge(2, 3);
        bfs(2);
    }
}

File: examples/csharp/Graph1.cs
using System;
using System.Collections.Generic;

class Graph1
{
    static void Main()
    {
        var adj = new Dictionary<int, List<int>>();
        adj[0] = new List<int> { 1, 2 };
        adj[1] = new List<int> { 2 };
        adj[2] = new List<int> { 0, 3 };
        adj[3] = new List<int> { 3 };
        BFS(adj, 2);
    }

    static void BFS(Dictionary<int, List<int>> adj, int start)
    {
        var visited = new HashSet<int>();
        var queue = new Queue<int>();
        queue.Enqueue(start);
        visited.Add(start);
        while (queue.Count > 0)
        {
            int v = queue.Dequeue();
            Console.Write(v + " ");
            foreach (int n in adj.GetValueOrDefault(v, new List<int>()))
                if (!visited.Contains(n)) { visited.Add(n); queue.Enqueue(n); }
        }
        Console.WriteLine();
    }
}
